flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/sinks/TableSinkUtils.scala

```diff
@@ -62,5 +68,18 @@ object TableSinkUtils {
           s"Query result schema: $srcSchema\n" +
           s"TableSink schema:    $sinkSchema")
     }
+    // check partitions are valid
+    sink match {
```

Comment:
Please check `partitionableTableSink.getPartitionFieldNames` is not empty first if `sinkOperation.getStaticPartitions` is not empty. Regarding to the javadoc, if `sink.getPartitionFieldNames` is empty, it is not a partitioned sink.